core/src/main/scala/org/apache/spark/SparkConf.scala

```diff
@@ -584,6 +679,32 @@ class SparkConf(loadDefaults: Boolean) extends Cloneable with Logging with Seria
       }
     }
 
+    val resources = getResources(false)
+    val resourceRequirements = getResourceRequirements()
+    val executorCores = get(EXECUTOR_CORES)
+    val taskCpus = get(CPUS_PER_TASK)
+    resourceRequirements.foreach { case (resourceName, resourceCount) =>
+      if (!resources.contains(resourceName)) {
```

Comment:
This should just be comparing task count to executor resource count (no addresses). Those are the 2 configs the user would specify.

could simplify this a bit by doing a getOrElse:

val executorResourceCount = resources.get(resourceName).getOrElse(
        throw new SparkException(...

Then you don't need the if/else and can use executorResourceCount in the if check below.